nodejs/eks/cluster.ts

```diff
@@ -184,38 +189,53 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     // Create the VPC CNI management resource.
     const vpcCni = new VpcCni(`${name}-vpc-cni`, kubeconfig, args.vpcCniOptions, { parent: parent });
 
-    // Create the instance role we'll use for worker nodes.
-    let instanceRole: pulumi.Output<aws.iam.Role>;
-    if (args.instanceRole) {
-        instanceRole = pulumi.output(args.instanceRole);
+    const instanceRoleMappings: RoleMapping[] = [];
+    let instanceProfile: aws.iam.InstanceProfile;
```

Comment:
Should this be `aws.iam.InstanceProfile | undefined` since there is a branch where it is unset?

I think that also avoids the strange `x! || undefined` later on?